src/SWP/Bundle/AnalyticsBundle/EventListener/KernelListener.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Template Engine Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\AnalyticsBundle\EventListener;
+
+use SWP\Bundle\AnalyticsBundle\Model\RequestMetric;
+use SWP\Bundle\AnalyticsBundle\Repository\RequestMetricRepository;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\Stopwatch\Stopwatch;
+use Symfony\Component\Stopwatch\StopwatchEvent;
+
+class KernelListener
```

Comment:
In that example you have multiple classes listening to the same kind of
events (doctrine events). Can the same classes be used to listen to kernel
events?

On Mon, Jun 13, 2016 at 12:56 PM, Paweł Mikołajczuk <
notifications@github.com> wrote:

> In src/SWP/Bundle/AnalyticsBundle/EventListener/KernelListener.php
> <https://github.com/superdesk/web-publisher/pull/78#discussion_r66773596>:
>
> > + * For the full copyright and license information, please see the
> > + * AUTHORS and LICENSE files distributed with this source code.
> > + *
> > + * @copyright 2016 Sourcefabric z.ú.
> > + * @license http://www.superdesk.org/license
> > + */
> > +namespace SWP\Bundle\AnalyticsBundle\EventListener;
> > +
> > +use SWP\Bundle\AnalyticsBundle\Model\RequestMetric;
> > +use SWP\Bundle\AnalyticsBundle\Repository\RequestMetricRepository;
> > +use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
> > +use Symfony\Component\HttpKernel\Event\GetResponseEvent;
> > +use Symfony\Component\Stopwatch\Stopwatch;
> > +use Symfony\Component\Stopwatch\StopwatchEvent;
> > +
> > +class KernelListener
>
> as it's not possible, afaik, to handle different kinds of events in the
> same class - it is with
> http://symfony.com/doc/current/cookbook/doctrine/event_listeners_subscribers.html
>
> Class name should be about functionality not about events, as you should
> be able to use it also for other events in future (not kernel but custom
> events).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/superdesk/web-publisher/pull/78/files/5fbb2cfc7cad6a7bafa68838427244bcec0eb8fa#r66773596>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ARBtFZu0AH3L9-0uYHsT5m0Ygv5iOa6Fks5qLTd3gaJpZM4Ix7A->
> .
>